Documentation/workflow/NativeDebuggingUsingAndroidStudio.md

```diff
@@ -0,0 +1,21 @@
+# Native Debugging of Mono using Android Studio
+
+With these steps it is possible to start a XA app with managed debugging in Visual Studio (on Mac and Windows) and native debugging of mono (with debugging symbols) using Android Studio.  
+
+Steps:
+1. Get the same github mono version that is in Android SDK
+2. Generate debug version of mono for android x86:
+    make -C sdks/builds build-android-x86 -j8 CONFIGURATION=debug
+3. Start x86 emulator
+4. Get root access on emulator:
+    adb root
+5. Find mono lib on emulator:
+    adb shell
+    find / -name libmonosgen-32bit-2.0.so
+6. Replace mono lib from path found using find command with mono debug lib :
+    adb push sdks/builds/android-x86-debug/mono/mini/.libs/libmonosgen-2.0.so /data/app/Mono.Android.DebugRuntime-oZUTbxntb9m9aK1M2CdL3g==/lib/x86/libmonosgen-32bit-2.0.so
```

Comment:
This assumes that the emulator is 32-bits.

Additionally, the aforementioned `bundle-v*.zip` contains *un*-`strip`d native libraries, e.g. [bundle-v21-h0f517d2a-Debug-Darwin-libzip=b95cf3f,llvm=349752c,mono=2264e1b.zip](https://jenkins.mono-project.com/view/Xamarin.Android/job/xamarin-android-debug/104/Azure/processDownloadRequest/xamarin-android-debug/xamarin-android/bin/Debug/bundle-v21-h0f517d2a-Debug-Darwin-libzip=b95cf3f,llvm=349752c,mono=2264e1b.zip) contains:

```bash
$ unzip -l /Users/jon/android-archives/bundle-v21-h5230fe52-Release-Darwin-libzip=b95cf3f,llvm=349752c,mono=89d7665.zip | grep libmonosg.*.d.so
 17919988  03-20-2019 12:12   lib/xamarin.android/xbuild/Xamarin/Android/lib/armeabi-v7a/libmonosgen-2.0.d.so
 16827072  03-20-2019 12:12   lib/xamarin.android/xbuild/Xamarin/Android/lib/arm64-v8a/libmonosgen-2.0.d.so
 18053048  03-20-2019 12:12   lib/xamarin.android/xbuild/Xamarin/Android/lib/x86/libmonosgen-2.0.d.so
 17186056  03-20-2019 12:12   lib/xamarin.android/xbuild/Xamarin/Android/lib/x86_64/libmonosgen-2.0.d.so
```

These `.d.so` files contain debug symbols.

Reply:
In case it's also useful to note, at least for the current version, the debug version of the library is also shipped in the product on macOS (though not on Windows):

*/Library/Frameworks/Xamarin.Android.framework/Versions/Current/lib/xamarin.android/xbuild/Xamarin/Android/lib/x86/libmonosgen-2.0.d.so*

Reply:
That won't be true in d16-2, which is a post-e83ba0ddc159d15abcfac1a5a4bce5a53e165d3f world.